kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/KafkaService.java

```diff
@@ -690,12 +693,20 @@ private InternalClusterMetrics calculateClusterMetrics(
     properties.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class);
     properties.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class);
     try (KafkaProducer<byte[], byte[]> producer = new KafkaProducer<>(properties)) {
-      final ProducerRecord<byte[], byte[]> producerRecord = serde.serialize(
+      ProducerRecord<byte[], byte[]> producerRecord = serde.serialize(
           topic,
           msg.getKey(),
           msg.getContent(),
           msg.getPartition()
       );
+      if (msg.getHeaders() != null && !msg.getHeaders().isEmpty()) {
+        producerRecord = new ProducerRecord<>(
+            producerRecord.topic(),
+            producerRecord.partition(),
+            producerRecord.key(),
+            producerRecord.value(),
+            createHeaders(msg.getHeaders()));
```

Comment:
no need for additional if - you can use main ProducerRecord constructor. (it is ok to pass null for headers to it)

Reply:
Will fix

Reply:
Fixed, please review